packages/@aws-cdk/aws-pipes-alpha/lib/source.ts

```diff
@@ -100,3 +103,48 @@ export interface ISource {
    */
   grantRead(grantee: IRole): void;
 }
+
+/**
+ * Sources that support a dead-letter target.
+ */
+export abstract class SourceWithDlq implements ISource {
+  /**
+   * The ARN of the source resource.
+   */
+  readonly sourceArn: string;
+  /**
+   * The dead-letter SQS queue or SNS topic.
+   */
+  readonly deadLetterTarget?: IQueue | ITopic;
+
+  constructor(sourceArn: string, deadLetterTarget?: IQueue | ITopic) {
+    this.sourceArn = sourceArn;
+    this.deadLetterTarget = deadLetterTarget;
+  }
+
+  abstract bind(pipe: IPipe): SourceConfig;
+  abstract grantRead(grantee: IRole): void;
+
+  /**
+   * Grants the pipe role permission to publish to the dead-letter target.
+   */
+  public grantDlqPush(grantee: IRole, deadLetterTarget?: IQueue | ITopic) {
```

Comment:
I would just name this `grantPush`. Since it's in this class, it's already established that it's acting on the DLQ.